core-web/libs/sdk/analytics/src/lib/plugin/dot-analytics.plugin.ts

```diff
@@ -0,0 +1,45 @@
+import { ANALYTICS_PAGEVIEW_EVENT } from '../shared/analytics.constants';
+import { sendAnalyticsEventToServer } from '../shared/analytics.http';
+import { DotAnalyticsConfig } from '../shared/analytics.model';
+import { createAnalyticsPageViewData } from '../shared/analytics.utils';
+
+/**
+ * The dotAnalytics plugin.
+ *
+ * @param {DotAnalyticsConfig} config - The analytics configuration.
+ * @returns {Object} - The dotAnalytics plugin.
+ */
+export const dotAnalyticsPlugin = (config: DotAnalyticsConfig) => {
+    let isInitialized = false;
+
+    return {
+        name: 'dot-analytics',
+        config,
+
+        initialize: ({ config }: { config: DotAnalyticsConfig }) => {
+            if (!config.server) {
+                throw new Error('DotAnalytics: Server URL is required');
```

Comment:
Would it be worth it to create a custom errror?

Reply:
Yes, but, maybe in the next iteration.